modules/custom/az_select_menu/src/Plugin/Block/AzSelectMenu.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace Drupal\az_select_menu\Plugin\Block;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Template\Attribute;
+use Drupal\Core\Url;
+use Drupal\menu_block\Plugin\Block\MenuBlock;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an extended Menu block.
+ *
+ * @Block(
+ *   id = "az_select_menu",
+ *   admin_label = @Translation("Quickstart select menu block"),
+ *   category = @Translation("Quickstart select menu block"),
+ *   deriver = "Drupal\menu_block\Plugin\Derivative\MenuBlock",
+ *   forms = {
+ *     "settings_tray" = "\Drupal\system\Form\SystemMenuOffCanvasForm",
+ *   },
+ * )
+ */
+class AzSelectMenu extends MenuBlock {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->menuParentFormSelector = $container->get('menu.parent_form_selector');
+    $instance->entityTypeManager = $container->get('entity_type.manager');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $config = $this->configuration;
+    $defaults = $this->defaultConfiguration();
+
+    $form = parent::blockForm($form, $form_state);
+    $form['az_select_menu'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Quickstart select menu options.'),
+    ];
+
+    $form['az_select_menu']['empty_option'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Include an empty option.'),
+      '#default_value' => $config['az_select_menu']['empty_option'],
+    ];
+
+    $form['az_select_menu']['empty_option_label'] = [
+      '#type' => 'textfield',
+      '#title' => t('Empty option label.'),
+      '#default_value' => $config['az_select_menu']['empty_option_label'],
+      '#states' => [
+        'visible' => [
+          ':input[name="settings][az_select_menu][empty_option"]' => ['checked' => TRUE],
```

Comment:
Did you already happen to try it like this?

```suggestion
          ':input[name="settings[az_select_menu][empty_option]"]' => ['checked' => TRUE],
```